### DISCUSSION TITLE
Difference between ST7789 and ST7796 displays ??

### DISCUSSION THREAD

**Discussion by Comalco** [Bodmer/TFT_eSPI#2837] [Compatible displays and setup files]
Hi,
A little off topic, but you folk seem to have had a lot of experience with the 7789 and 7796 displays.
I have a small project running happily using the 7789 (2inch 320x200) on an ESP32 using SPI.
I spotted a 7796 (4inch) and bought a couple.
I have the two displays (7789 and 7796) wired in parallel, which should be fine because I am not using MISO on either
I have amended the driver I am using to accept the 4inch 480x320 and my code runs without throwing errors.
BUT, not matter what I try, including trying a second display, I cannot get anything to appear on the 7796 displays.
Backlight works, and they dont have that troublesome diode some earlier 7796's had.
I can unplug the 7789 so only the 7796 is connected, but still no display of anything!
I have checked wiring many times and I am confident everything is connected correctly.

Looking at the displays' datasheets, the command opcodes and SPI timing and logic pin usage all appears to be the same between the 7789 and 7786 - with, from what I can tell, only the WxH being different.

I am using Micropython and Russ Hughes ST7789.py driver on GitHub.  https://github.com/russhughes/st7789py_mpy
As mentioned, I have edited this to accept the 7796 as larger HxW size.
 
As I'm a newbie to ESP32s, I am at a bit of a dead end to understand what the difference might be between the 7789 working perfectly (writing text, lines, circles or rectangles), and the 7796 not showing anything.

I am assuming there is something very subtle as a difference between the 7789 and 7796?  
Could anyone share any hints on what I might be missing please?   TIA.

**Comment by Bodmer** [comment]
At power on the displaycontrol registers are set to certain values, these values are different for the two displays, so an ST7789 driver will not work with an ST7796 display.

**Reply by Comalco** [reply]
Thanks.  I will read up the datasheets again.  A few writes in an init routine should put the 7796 into a similar set as the 7789?.  
A good excuse for me to try, and to learn in the process.

**Reply by Comalco** [reply]
Bodmer,
I have gone through the display control registers and I cannot identify any differences in MODCTL (36h), or any of the other registers.
There is a small difference in CASET (2Ah) but I dont believe this will prevent something on the display when drawing lines, rectangles, etc.
Could you please assist and give me a clue where I need to look ??   TIA

**Reply by Comalco** [reply]
Bodmer,
I ported your drivers over to Micrpython - making both a 7789 version and a 7796 version.
The 7789 version seems to work flawlessly on the 7789 display, but nothing appears on the 7796 display (as might be expected).
The 7796 driver still displays on the 7789 display, but I still get nothing on the 7796.
Nothing seems to create anything on the 7796 display.
My creations attached FYI and comments welcomed please.  TIA 
[st7789py.txt](https://github.com/Bodmer/TFT_eSPI/files/12551621/st7789py.txt)
[st7796py.txt](https://github.com/Bodmer/TFT_eSPI/files/12551622/st7796py.txt)
